mkosi/__init__.py

```diff
@@ -6592,7 +6630,22 @@ def setup_ssh(
         return None
 
     if args.distribution in (Distribution.debian, Distribution.ubuntu):
-        unit = "ssh"
+        unit = "ssh.socket"
+
+        if args.ssh_port != 22:
+            override_dir = root / "etc/systemd/system/ssh.socket.d"
+            override_dir.mkdir(mode=0o755, parents=True, exist_ok=True)
+            with open(override_dir / "port.conf", "w") as s:
+                s.write(
+                    dedent(
+                        f"""\
+                        [Socket]
+                        ListenStream=
+                        ListenStream={args.ssh_port}
+                        """
+                    )
+                )
+            os.chmod(override_dir / "port.conf", 0o644)
```

Comment:
With #853 landed you can use `add_dropin_config` here.

Reply:
ok, rebased and updated